lib/mix/lib/mix.ex

```diff
@@ -352,6 +352,10 @@ defmodule Mix do
     * `MIX_INSTALL_DIR` *(since v1.12.0)* - specifies directory where `Mix.install/2` keeps
        install cache
 
+    * `MIX_DISABLE_LOCK` - disables mix compilation locking. While not recommended, this may be
```

Comment:
My only concern is that the name sounds related to `mix.lock`, so alternative ideas are welcome. Though it's going to be rarely used, so maybe that's fine.

Reply:
maybe `MIX_DISABLE_CONCURRENCY_LOCKS`? I am not sure I like `_DISABLE_` in the name though. It could also be `MIX_CONCURRENCY_LOCKS=0` or `MIX_CONCURRENCY_LOCKS=false`?

Reply:
`MIX_COMPILER_LOCK` maybe? It’s not affecting concurrency “in general” just turning off the new compiler lock.

Reply:
It applies to deps management as well. What about `MIX_OS_CONCURRENCY_LOCK`?

Reply:
`MIX_OS_CONCURRENCY_LOCK` sounds good, I'd say in this case longer is actually better :D I'm generally not a fan `_DISABLE_` either, but I saw that the other env vars (`MIX_DEBUG`, `MIX_QUIET`) are opt-ins. If you are fine with 0/false then definitely works for me!